File: src/main/java/com/are/springlearning/bean/ResponseBodyBean.java
package com.are.springlearning.bean;

import com.fasterxml.jackson.databind.ObjectMapper;

public class ResponseBodyBean {

    private int state = 200;

    private String message = "success";

    private Object body = null;

    public ResponseBodyBean(){
        final ObjectMapper mapper = new ObjectMapper();
        body = mapper.createObjectNode();
    }

    public ResponseBodyBean(int state, String message, Object body){
        setState(state);
        setMessage(message);
        setBody(body);
    }

    public int getState() {
        return state;
    }

    public void setState(int state) {
        this.state = state;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Object getBody() {
        return body;
    }

    public void setBody(Object body) {
        this.body = body;
    }
}

File: src/main/java/com/are/springlearning/bean/ResponseBodyBean.kt
package com.are.springlearning.bean

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.ObjectMapper

class ResponseBodyBean {

    @JsonProperty
    private var state = 0

    @JsonProperty
    private var message = "success"

    @JsonProperty
    private var body: Any? = ObjectMapper().createObjectNode()

    constructor() {

    }

    constructor(body: Any){
        this.body = body
    }

    constructor(state: Int, message: String){
        this.state = state
        this.message = message
    }

    constructor(state: Int, message: String, body: Any) {
        this.state = state
        this.message = message
        this.body = body
    }

}
